src/sdk/pynni/nni/compression/torch/utils/shape_dependency.py

```diff
@@ -185,6 +185,49 @@ def dependency_sets(self):
             d_sets.append(tmp_set)
         return d_sets
 
+class InputChannelDependency(ChannelDependency):
+    def __init__(self, model, dummy_input=None, traced_model=None):
+        super(InputChannelDependency, self).__init__(model, dummy_input, traced_model)
+
+    def _get_following_convs(self, tensor):
+        queue = []
+        following_convs = []
+        queue.extend(self.graph.input_to_node[tensor])
+        while queue:
+            curnode = queue.pop(0)
+            if curnode.op_type == 'Conv2d' or curnode.op_type == 'Linear':
+                # find the first met conv
+                following_convs.append(curnode.name)
+                continue
+            successors = self.graph.find_successors(curnode.unique_name)
+            successors = [self.graph.name_to_node[name] for name in successors]
```

Comment:
seems you only handle `Conv2d` and `Linear`. Is it possible that there is an op type between `tensor` and one following op, and this op type blocks the spread of shape inference?

Reply:
Yes, reshape operations (such as `view` ) will change the shape of the tensor and may truncate the dependency relation.

Reply:
updated~